### DISCUSSION TITLE
Flow runs are SLOW.

### DISCUSSION THREAD

**Discussion by kensouchen** [PrefectHQ/prefect#4972] [Q&A] [answered]
Hi team,

Im trying to implement prefect into my script, but somehow a flow run is too slow.

```python

@task
def foo():
    return 1

with Flow("Test") as myflow:
    resp = foo()


%time myflow.run()
CPU times: user 36.1 ms, sys: 981 µs, total: 37.1 ms
Wall time: 36.4 ms

# Running this flow 100 times, takes 3.3 seconds....which makes no sense...
%time [myflow.run() for _ in range(100)]
CPU times: user 3.3 s, sys: 5.06 ms, total: 3.31 s
Wall time: 3.33 s


# Running task itself is quick.
%time foo.run()
CPU times: user 6 µs, sys: 0 ns, total: 6 µs
Wall time: 9.06 µs
```

Run a flow is usually way much slower than simple run the series of tasks.

When profiling a flow run, I saw `python-box` being the main reason of slowness.

-------

In my real code, all my task input/outputs are serializable. All my task runs add up to use 0.03 second, but a flow run takes ~0.3 seconds, which made it 10 times slower.

Is this slowness expected? or am I triggering my flow in the right way?

**Comment by zanieb** [comment, marked as answer]
Hi @kensouchen -- this is expected. Flow runs are not intended to operate on a sub-second scale. If that is a requirement, you're better off using lower level tooling like dask.

**Reply by kensouchen** [reply]
Thanks, make sense.
